core/src/test/java/org/apache/iceberg/TestDeleteFiles.java

```diff
@@ -300,6 +322,34 @@ public void testDeleteCaseSensitivity() {
         statuses(Status.DELETED));
   }
 
+  @Test
+  public void testDeleteFilesOnIndependentBranches() {
+    String testBranch = "testBranch";
+    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).appendFile(FILE_C).commit();
+    Snapshot initialSnapshot = table.currentSnapshot();
+    // Delete A on test branch
+    table.newDelete().deleteFile(FILE_A).toBranch(testBranch).commit();
+    Snapshot testBranchTip = table.snapshot(testBranch);
+
+    // Delete B and C on main
+    table.newDelete().deleteFile(FILE_B).deleteFile(FILE_C).commit();
+    Snapshot delete2 = table.currentSnapshot();
+
+    // Verify B and C on testBranch
+    validateManifestEntries(
+        testBranchTip.allManifests(FILE_IO).get(0),
```

Comment:
It's much better to use something like `Iterables.getOnlyElement` rather than `.get(0)` to validate your assumptions about there being only one manifest.